sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -2242,6 +2242,29 @@ abstract class SQLQuerySuiteBase extends QueryTest with SQLTestUtils with TestHi
     }
   }
 
+  test("SPARK-32889 ORC table column name supports special characters like $ eg.") {
+    Seq("$").foreach { name =>
+      Seq("ORC").foreach { source =>
+        withSQLConf("spark.sql.orc.column.allowSpecialChar" -> "true") {
+          Seq(s"CREATE TABLE t32889(`col$name` INT) USING $source",
+              s"CREATE TABLE t32889 STORED AS $source AS SELECT 1 `col$name`",
+              s"CREATE TABLE t32889 USING $source AS SELECT 1 `col$name`",
+              s"CREATE TABLE t32889(`col$name` INT) USING hive OPTIONS (fileFormat '$source')")
+          .foreach { command =>
+            withTable("t32889") {
```

Comment:
could you extract the table name as a `val`, and give a more meaningful name?

Reply:
hi wzhfy. this is jira number. i saw other uts use this way.